core/src/main/java/org/hisp/dhis/android/core/event/search/EventQueryCollectionRepository.kt

```diff
@@ -0,0 +1,315 @@
+            scope.toBuilder().dataFilters(filters).build()
+        }
+    }
+
+    fun orderByEventDate(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.EVENT_DATE)
+    }
+
+    fun orderByDueDate(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.DUE_DATE)
+    }
+
+    fun orderByCompleteDate(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.COMPLETED_DATE)
+    }
+
+    fun orderByCreated(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.CREATED)
+    }
+
+    fun orderByLastUpdated(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.LAST_UPDATED)
+    }
+
+    fun orderByOrganisationUnitName(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.ORGUNIT_NAME)
+    }
+
+    fun orderByTimeline(): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.TIMELINE)
+    }
+
+    fun orderByDataElement(dataElement: String?): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return orderConnector(EventQueryScopeOrderColumn.dataElement(dataElement))
+    }
+
+    private fun orderConnector(
+        col: EventQueryScopeOrderColumn,
+    ): EqFilterConnector<EventQueryCollectionRepository, OrderByDirection> {
+        return connectorFactory.eqConnector { direction ->
+            val order: MutableList<EventQueryScopeOrderByItem> = ArrayList(scope.order())
+            order.add(EventQueryScopeOrderByItem.builder().column(col).direction(direction).build())
+            scope.toBuilder().order(order).build()
+        }
+    }
+
+    override fun uid(uid: String?): EventObjectRepository {
+        return eventCollectionRepository.uid(uid)
+    }
+
+    override fun getUids(): Single<List<String>> {
+        return eventCollectionRepository.getUids()
+    }
+
+    override fun blockingGetUids(): List<String> {
+        return eventCollectionRepository.blockingGetUids()
+    }
+
+    override fun get(): Single<List<Event>> {
+        return eventCollectionRepository.get()
+    }
+
+    override fun blockingGet(): List<Event> {
+        return eventCollectionRepository.blockingGet()
+    }
+
+    override fun getPaged(pageSize: Int): LiveData<PagedList<Event>> {
+        return eventCollectionRepository.getPaged(pageSize)
+    }
+
+    override fun getPagingData(pageSize: Int): Flow<PagingData<Event>> {
+        return eventCollectionRepository.getPagingData(pageSize)
+    }
+
+    fun getDataSource(): DataSource<Event, Event> {
```

Comment:
This property should be something like:
```
val dataSource: DataSource<Event, Event>
        get() = eventCollectionRepository.dataSource
```
This is just to avoid breaking changes in the API